service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -303,11 +303,16 @@ func (t *transferQueueActiveTaskExecutor) processCloseExecution(
 	}
 	executionInfo := mutableState.GetExecutionInfo()
 	executionState := mutableState.GetExecutionState()
-	completionEvent, err := mutableState.GetCompletionEvent(ctx)
-	if err != nil {
-		return err
+	var completionEvent *historypb.HistoryEvent // needed to report close event to parent workflow
+	replyToParentWorkflow := mutableState.HasParentExecution() && executionInfo.NewExecutionRunId == ""
+	if replyToParentWorkflow {
+		// only load close event if needed.
+		completionEvent, err = mutableState.GetCompletionEvent(ctx)
+		if err != nil {
+			return err
+		}
+		replyToParentWorkflow = replyToParentWorkflow && !IsTerminatedByResetter(completionEvent)
```

Comment:
Why this condition is separated from the other two?

Reply:
Because this condition require the close event and I want to avoid load close event if it is not necessary.

Reply:
o right. Missed that it depends on the close event. 

hmmm, I guess we also need the same change for standby transfer executor.